src/sysinfo.rs

```diff
@@ -5,33 +5,187 @@ use crate::systemapi;
+    attrs.push_s("node", node_info.node.clone());
+    attrs.push_s("os_name", system.get_os_name());
+    attrs.push_s("os_release", system.get_os_release());
+    let mut cores = output::Array::new();
+    for k in &node_info.cores {
+        let mut c = output::Object::new();
+        c.push_i("index", k.logical_index);
+        c.push_i("physical", k.physical_index);
+        c.push_s("model", k.model.clone());
+        cores.push_o(c);
+    }
+    attrs.push_a("cores", cores);
+    attrs.push_i("memory", node_info.mem_kb);
+    attrs.push_s("description", node_info.description.clone());
+    // attrs.push_s("topo_svg") // FIXME
+    let gpu_info = layout_card_info(&node_info, true);
+    if gpu_info.len() > 0 {
+        attrs.push_a("cards", gpu_info);
+    }
+    // software.push_s("key")  // FIXME
+    // software.push_s("name") // FIXME
+    // software.push_s("version") // FIXME
+    // attrs.push_a("software") // FIXME
+    data.push_o("attributes", attrs);
+    envelope.push_o("data", data);
+    envelope
+}
+
+fn layout_error_newfmt(system: &dyn systemapi::SystemAPI, error: String) -> output::Object {
+    let mut envelope = output::newfmt_envelope(system);
+    envelope.push_a("errors", output::newfmt_one_error(system, error));
+    envelope
+}
+
+// Old JSON/CSV format - this is flattish (to accomodate CSV) and has some idiosyncratic field names.
+
+//+oldnames
+fn layout_sysinfo_oldfmt(system: &dyn systemapi::SystemAPI, node_info: NodeInfo) -> output::Object {
+    let gpu_info = layout_card_info(&node_info, false);
+    let mut sysinfo = output::Object::new();
+    sysinfo.push_s("version", system.get_version());
+    sysinfo.push_s("timestamp", system.get_timestamp());
+    sysinfo.push_s("hostname", system.get_hostname());
+    sysinfo.push_s("description", node_info.description);
+    sysinfo.push_i("cpu_cores", node_info.cpu_cores);
+    sysinfo.push_i("mem_gb", node_info.mem_kb / (1024 * 1024));
+    if node_info.gpu_cards != 0 {
+        sysinfo.push_i("gpu_cards", node_info.gpu_cards);
+        if node_info.gpumem_kb != 0 {
+            sysinfo.push_i("gpumem_gb", node_info.gpumem_kb / (1024 * 1024));
+        }
+        if gpu_info.len() > 0 {
+            sysinfo.push_a("gpu_info", gpu_info);
+        }
+    }
+    sysinfo
+}
 
-pub fn compute_sysinfo(system: &dyn systemapi::SystemAPI) -> output::Object {
-    try_compute_sysinfo(system).unwrap_or_else(|e: String| error_packet(system, e))
+fn layout_error_oldfmt(system: &dyn systemapi::SystemAPI, error: String) -> output::Object {
+    let mut sysinfo = output::Object::new();
+    sysinfo.push_s("version", system.get_version());
+    sysinfo.push_s("timestamp", system.get_timestamp());
+    sysinfo.push_s("hostname", system.get_hostname());
+    sysinfo.push_s("error", error);
+    sysinfo
+}
+//-oldnames
+
+// Old and new formats layout the cards the same way but use different field names in a number of
+// cases.
```

Comment:
This code is brittle wrt the test that greps for strings and compares them to known fields.  It will work fine so long as the new names are used in the "if" part of the tests below and nothing goes over a line break.  May note that here.  Long-term the old-fmt code will go away I guess.